libqtile/core/manager.py

```diff
@@ -142,6 +143,7 @@ def noop(qtile):
         self.config.mouse += (Click([], "Button1", lazy.function(noop), focus="after"),)
 
     def dump_state(self, buf):
+        del(self.chord_stack)  # TODO: is there a better way to avoid errors?
```

Comment:
What is this for? Could you elaborate on what error you get and need to avoid?

Reply:
When qtile is restarted, the current state is dumped. But the `KeyChord` on the stack do not support serialisation such that restoring the state would actually work. And, in fact, if the user changes keybindings in the config and then restarts qtile, he/she expects to be in the root mode (not in the chord mode they might have been in previously - that would be unintuitive, complicated to implement correctly and not an important feature at all). Additionally, `current_chord` is an internal shortcut to check if we stay in a mode or if we need to call `cmd_ungrab_chord()`. Persisting one of `chord_stack` and `current_chord` would result in inconsistencies. See also: [my comment here](https://github.com/qtile/qtile/issues/2241#issuecomment-786790819).
Actually, you made me think about it so that I might want to revisit the state topic: If `current_chord` is dumped/restored, I might need to un-persist that one too.

Reply:
Do we need to store either? Currently neither is dumped, and I wouldn't expect either to persist across restarts.

Reply:
Oh, I see. This line is really unnecessary, right? :rofl:

Reply:
I believe so :)

Reply:
Removed that dummy line, too...